ProjFS.Mac/PrjFSKextTests/KextMockUtilities.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+
+class FunctionTypeCallRecorderBase
+{
+public:
+    virtual void Clear() = 0;
+    virtual ~FunctionTypeCallRecorderBase() {}
+};
+
+template <typename R, typename... ARGS>
+    class FunctionTypeCallRecorder : public FunctionTypeCallRecorderBase
+{
+    typedef R (*FunctionPointerType)(ARGS...);
+    struct FunctionCall
+    {
+        // TODO: add argument values
+        uint64_t callSequenceNumber;
+    };
+    
+    typedef std::unordered_multimap<FunctionPointerType, FunctionCall> RecordedCallMapType;
```

Comment:
Another minor naming suggestion, I think you could drop the "Type" from `RecordedCallMapType` (i.e. `RecordedCallMapType` -> `RecordedCallMap`).

I saw that our `typedef std::unordered_map` typedefs don't use "Type" in the type name.